l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -264,27 +293,43 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         whenNotPaused
         returns (L2TransactionRequestTwoBridgesInner memory request)
     {
-        require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
+        if (l2BridgeAddress[_chainId] == address(0)) {
+            revert L2BridgeNotSet(_chainId);
+        }
 
         (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
             _data,
             (address, uint256, address)
         );
-        require(_l1Token != L1_WETH_TOKEN, "ShB: WETH deposit not supported");
-        require(BRIDGE_HUB.baseToken(_chainId) != _l1Token, "ShB: baseToken deposit not supported");
+        if (_l1Token == L1_WETH_TOKEN) {
+            revert TokenNotSupported(L1_WETH_TOKEN);
+        }
+        if (BRIDGE_HUB.baseToken(_chainId) == _l1Token) {
+            revert TokenNotSupported(_l1Token);
+        }
 
         uint256 amount;
         if (_l1Token == ETH_TOKEN_ADDRESS) {
             amount = msg.value;
-            require(_depositAmount == 0, "ShB wrong withdraw amount");
+            if (_depositAmount != 0) {
+                revert WithdrawIncorrectAmount();
+            }
         } else {
-            require(msg.value == 0, "ShB m.v > 0 for BH d.it 2");
+            if (msg.value != 0) {
+                revert NonEmptyMsgValue();
+            }
             amount = _depositAmount;
 
             uint256 withdrawAmount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _depositAmount);
-            require(withdrawAmount == _depositAmount, "5T"); // The token has non-standard transfer logic
+            // The token has non-standard transfer logic
+            if (withdrawAmount != _depositAmount) {
+                revert WithdrawIncorrectAmount();
```

Comment:
In similar places the arguments were provided in the error:

```
                revert ValueMismatch(_amount, amount);
```
Consider providing them here as well